cluster/docker-cluster/util.sh

```diff
@@ -0,0 +1,218 @@
+set -e
+
+
+# Verify requests
+# 
+# Assumed vars:
+#   MASTER
+#   NODES
+function verify-prereqs {
+  local rc
+
+  rc=0
+  ssh-add -L 1> /dev/null 2> /dev/null || rc="$?"
+  # "Could not open a connection to your authentication agent."
+  if [[ "${rc}" -eq 2 ]]; then
+    eval "$(ssh-agent)" > /dev/null
+    trap-add "kill ${SSH_AGENT_PID}" EXIT
+  fi
+
+  rc=0
+  ssh-add -L 1> /dev/null 2> /dev/null || rc="$?"
+  # "The agent has no identities."
+  if [[ "${rc}" -eq 1 ]]; then
+    # Try adding one of the default identities, with or without passphrase.
+    ssh-add || true
+  fi
+  # Expect at least one identity to be available.
+  if ! ssh-add -L 1> /dev/null 2> /dev/null; then
+    echo "Could not find or add an SSH identity."
+    echo "Please start ssh-agent, add your identity, and retry."
+    exit 1
+  fi
+
+  # Make sure env is properly set
+  : ${MASTER?"Need to set MASTER"}
+  : ${NODES?"Need to set NODES"}
+}
+
+# Verify cluster
+# 
+# Assumed vars:
+#   MASTER
+#   NODES
+#   SSH_OPTS
+function validate-cluster {
+  sleep 5 # For now we just sleep to wait the world
+  for node in $NODES
+  do
+    {
+      if [ "$node" == $MASTER ]; then
+        ssh $SSH_OPTS $node "bash ~/docker-cluster/provision/verify.sh master"
+      else
+        ssh $SSH_OPTS $node "bash ~/docker-cluster/provision/verify.sh node"
+      fi
+    }
+  done
+
+  echo
+  echo "Kubernetes-in-docker cluster is deployed.  The master should be running at:"
+  echo
+  echo "  http://${MASTER_IP}:8080"
+}
+
+
+# Instantiate a kubernetes cluster on docker-cluster
+#
+# Assumed vars:
+#   MASTER
+#   NODES
+# Vars set:
+#   KUBE_ROOT
+#   NUM_MINIONS
+function kube-up() {
+  KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+  source "docker-cluster/config-default.sh"
+
+  generate_env
+
+  NUM_MINIONS=0
+  for node in $NODES
+  do
+    {
+      if [ "$node" == $MASTER ]; then
+        if [[ -z $NODE_ONLY ]]; then
+          provision-node-master $MASTER_IP
+        fi
+      else
+        provision-node ${node#*@}
+      fi
+      NUM_MINIONS=$((NUM_MINIONS+1))
+    }
+  done
+  wait
+
+  export NUM_MINIONS=$NUM_MINIONS
+
+  echo "Deploy Complete!"
+}
+
+# Generate ENV file for nodes to use
+#
+# Assumed vars:
+#   in config-default.sh
+function generate_env(){
+  cat <<EOF > docker-cluster/kube-config/node.env
+# Do not modify!
+# This file is generated by docker-cluster/util.sh#generate_env()
+export FLANNEL_VERSION=$FLANNEL_VERSION
+export ETCD_VERSION=$ETCD_VERSION
+export K8S_VERSION=$K8S_VERSION
+export NODES="$NODES"
+export MASTER=$MASTER
+export MASTER_IP=$MASTER_IP
+export SSH_OPTS="$SSH_OPTS"
+export FLANNEL_NET=$FLANNEL_NET
+EOF
+  cat docker-cluster/kube-config/node.env
+
+  read -p "Are you sure? (y|n) " -n 1 -r
+  echo
+  if [[ ! $REPLY =~ ^[Yy]$ ]]
+  then
+      echo "Abort!"
+      exit 1
+  fi
+
+}
+
+# Provison master
+#
+# Assumed vars:
+#   MASTER_IP
+#   MASTER
+#   SSH_OPTS
+function provision-node-master() {
```

Comment:
To me work "provision" suggests that actually create the node. I'd expect that you provision a node if create a new VM in GCE/AWS. Here you are just configuring it. How about names ```configure-master()``` or ```deploy-to-master()```?

The same below.